src/components/HdToggle.vue

```diff
@@ -0,0 +1,243 @@
+<template>
+  <div
+    :class="{
+      'hd-toggle--is-open': open,
+      'hd-toggle--is-using-mouse': isUsingMouse,
+    }"
+    class="hd-toggle"
+    @keydown="setUsingMouse(false)"
+    @mousedown="setUsingMouse(true)"
+  >
+    <button
+      :disabled="!canBeToggled"
+      class="hd-toggle__control"
+      type="button"
+      @click="toggleOpen"
+    >
+      {{ title }}
+    </button>
+    <div
+      ref="body"
+      :style="{
+        maxHeight: `${maxHeight}px`,
+        transitionDuration: `${transitionDuration}ms`,
+      }"
+      class="hd-toggle__body"
+    >
+      <slot />
+    </div>
+  </div>
+</template>
+
+<script>
+import { OnResizeService } from 'homeday-blocks';
+
+export default {
+  name: 'HdToggle',
+  props: {
+    title: {
+      type: String,
+      default: '',
+    },
+    canBeToggled: {
+      type: Boolean,
+      default: true,
+    },
+    open: {
+      type: Boolean,
+      default: true,
+    },
+    transitionDuration: {
+      type: Number,
+      default: 300,
+    },
+  },
+  data() {
+    return {
+      // bodyHeight is a string only initially, will be set to a Number
+      // as soon as it is mounted
+      bodyHeight: '100%',
+      isUsingMouse: false,
+      internalFocusRemoved: false,
+    };
+  },
+  computed: {
+    maxHeight() {
+      if (this.open) {
+        return this.bodyHeight;
+      }
+
+      return 0;
+    },
+  },
+  watch: {
+    open: {
+      handler(open) {
+        this.ensureBodyScrolledToTop();
+
+        this.$nextTick(() => {
+          if (open) {
+            this.enableInternalFocus();
+          } else {
+            this.disableInternalFocus();
+          }
+        });
+      },
+      immediate: true,
+    },
+  },
+  mounted() {
+    this.$nextTick(this.calculateBodyHeight);
+    this.addResizeEvents();
+  },
+  beforeDestroy() {
+    this.removeResizeEvents();
+  },
+  methods: {
+    // Encapsulated in a method to be able to mock it in the tests
+    getScrollHeight(el) {
+      return el.scrollHeight;
+    },
+    toggleOpen() {
+      if (this.canBeToggled === false) {
+        return;
+      }
+
+      this.$emit('toggle', !this.open);
+    },
+    calculateBodyHeight() {
+      if (!this.$refs.body) {
+        return;
+      }
+
+      this.bodyHeight = this.getScrollHeight(this.$refs.body);
+    },
+    ensureBodyScrolledToTop() {
+      if (!this.$refs.body) {
+        return;
+      }
+
+      this.$refs.body.scrollTop = 0;
+    },
+    disableInternalFocus() {
```

Comment:
Focus is not covered with tests, do you think it would make sense to cover it? If you decided not to, as this is internal behavior, I think that's valid as well. :+1: